shell/platform/windows/flutter_desktop_messenger.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_FLUTTER_DESKTOP_MESSENGER_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_FLUTTER_DESKTOP_MESSENGER_H_
+
+#include <atomic>
+#include <mutex>
+
+#include "flutter/shell/platform/common/public/flutter_messenger.h"
+
+namespace flutter {
+
+class FlutterWindowsEngine;
+
+class FlutterDesktopMessenger {
+ public:
+  FlutterDesktopMessenger() = default;
+
+  /// Convert to FlutterDesktopMessengerRef.
+  FlutterDesktopMessengerRef ToRef() {
+    return reinterpret_cast<FlutterDesktopMessengerRef>(this);
+  }
```

Comment:
Why are these part of the class rather than in `flutter_window.cc` with the others?

The intended model here is that there's a coherent C++ interface that all internal logic is developed against, and then `flutter_windows.*` is a minimal layer over that for the sole purpose of creating a DLL interface (which has to be C). These opaque handles are purely a function of the C API layer, which is why the other functions are in `flutter_window.cc`; the C++ classes shouldn't know about the C layer when we can avoid it since that's a layering inversion. (In pattern terms I would view the C layer as basically a Facade, if that helps.)

Reply:
They were put there as the most convenient location.  They can't live in `flutter_window.cc` because they are also used in flutter_windows_engine.cc because of `BinaryMessengerImpl` and `IncomingMessageDispatcher` using `FlutterDesktopMessengerRef`.  I considered changing those interfaces but decided to do the least amount of change possible.

Reply:
Ah, right. Could you put a TODO on these then to move them to flutter_windows.cc functions and remove the internal usage of `FlutterDesktopMessengerRef`?

Reply:
done, added issue: https://github.com/flutter/flutter/issues/115021